test/utils/harness/harness.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package harness
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/glog"
+)
+
+// Harness adds some functionality to testing.T, in particular resource cleanup.
+// It embeds testing.T, so should have the same signature.
+//
+// Example usage:
+// ```
+// func MyTest(tt *testing.T) {
+//   t := harness.For(tt)
+//   defer t.Close()
+//   ...
+// }
+// ```
+type Harness struct {
```

Comment:
nit: Maybe make `Harness` an interface and `TempDirCleanupHarness` a concrete type of `Harness`? That way folks can implement different types of Harnesses?

Reply:
So the reason not to is because Harness embeds `testing.T`, so we can use it more interchangeably with existing code that wants a `testing.T`.  I do agree that once we've got it so that everyone is using the harness, we should move to an interface!